src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,231 @@
+    TableResult tableResult = getTableNamesFromDataSet();
+    Iterable<FieldValueList> rows = tableResult.iterateAll();
+
+    for (FieldValueList row : rows) {
+      FieldValue fieldValue = row.get(0);
+      String currentTableName = fieldValue.getStringValue();
+
+      if (currentTableName.equals(tableName)) {
+        return currentTableName;
+      }
+    }
+
+    return null; // Table not found
+  }
+
+  public static boolean compareSalesforceAndJsonData(List<JsonObject> salesforceData, List<JsonObject> bigQueryData,
+                                                     String tableName) {
+    boolean result = false;
+    if (bigQueryData == null) {
+      Assert.fail("bigQueryData is null");
+      return result;
+    }
+
+    int columnCountTarget = 0;
+    if (salesforceData.size() > 0) {
+      columnCountTarget = salesforceData.get(0).entrySet().size();
+    }
+    // Get the column count of the first JsonObject in bigQueryData
+    int jsonObjectIdx = 0;
+    int columnCountSource = 0;
+    if (bigQueryData.size() > 0) {
+      columnCountSource = bigQueryData.get(jsonObjectIdx).entrySet().size();
+    }
+
+    BigQuery bigQuery = BigQueryOptions.getDefaultInstance().getService();
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+
+    // Build the table reference
+    TableId tableRef = TableId.of(projectId, dataset, tableName);
+
+    // Get the table schema
+    Schema schema = bigQuery.getTable(tableRef).getDefinition().getSchema();
+
+    // Iterate over the fields
+    int currentColumnCount = 1;
+    while (currentColumnCount <= columnCountSource) {
+      for (Field field : schema.getFields()) {
+        String columnName = field.getName();
+        String columnType = field.getType().toString();
+        String columnTypeName = field.getType().getStandardType().name();
+
+        switch (columnType) {
+
+          case "BOOLEAN":
+            boolean sourceAsBoolean = salesforceData.get(0).get(columnName).getAsBoolean();
+            boolean targetAsBoolean = bigQueryData.get(jsonObjectIdx).get(columnName).getAsBoolean();
+            Assert.assertEquals("Different values found for column : %s", sourceAsBoolean, targetAsBoolean);
+            break;
+
+          case "FLOAT":
+            Double sourceVal = salesforceData.get(0).get(columnName).getAsDouble();
+            Double targetVal = bigQueryData.get(jsonObjectIdx).get(columnName).getAsDouble();
+            Assert.assertEquals(String.format("Different values found for column: %s", columnName), 0,
+                                sourceVal.compareTo(targetVal));
+            break;
+
+          case "TIMESTAMP":
+            OffsetDateTime sourceTimestamp = OffsetDateTime.parse(salesforceData.get(0).get(columnName).getAsString(),
+                                                                  DateTimeFormatter.ofPattern(
```

Comment:
Correct the code formating in this method. For ex :  (2 space indentation)

ffsetDateTime sourceTimestamp = OffsetDateTime.parse(salesforceData.get(0).get(columnName).getAsString(),
  DateTimeFormatter.ofPattern(....);

Reply:
done